tekton/resources/cd/configmap-template.yaml

```diff
@@ -75,12 +75,21 @@ spec:
             set -ex
             kubectl get configmap -n $(inputs.params.namespace) \
               $(inputs.params.configMapName) -o template \
-              --template='{{ index .data "$(inputs.params.configMapKey)" }}' > /workspace/$(inputs.params.configMapKey)
+              --template='{{ index .data "$(inputs.params.configMapKey)" }}' > \
+               /workspace/$(inputs.params.configMapKey) || \
+               rm /workspace/$(inputs.params.configMapKey)
```

Comment:
i dont understand what kind of values are expected for `configMapKey`, maybe it would make sense to use the description field for this one?

Reply:
Example: `config.yaml` is the configMapKey.
```
apiVersion: v1
data:
  config.yaml: "plank:\n  job_url_template: 'https://tekton-releases.appspot.com/build/tekton-prow/{{if
    or (eq .Spec.Type \"presubmit\") (eq .Spec.Type \"batch\")}}pr-logs/pull{{with
    .Spec.Refs}}/{{.Org}}_{{.Repo}}{{end}}{{else}}logs{{end}}{{if eq .Spec.Type \"presubmit\"}}/{{with
````

Example: `labels.yaml` is the configMapKey:

```
apiVersion: v1
data:
  labels.yaml: |
    # default: global configuration to be applied to all repos
    # repos: list of repos with specific configuration to be applied in addition to default
    #   labels: list of labels - keys for each item: color, description, name, target, deleteAfter, previously
    #     deleteAfter: 2006-01-02T15:04:05Z (rfc3339)
    #     previously: list of previous labels (color name deleteAfter, previously)
    #     target: one of issues, prs, or both (also TBD)
    #     addedBy: human? prow plugin? other?
    ---
    default:
```

Eventually we could switch this to the config map generator of kustomize, for now this is just a refactoring of things that we already do into one single job.

Reply:
Thanks for explaining - i can see that `configMapKey` in the trigger template does have a description field, but i guess one has to understand how keys in configmaps work to understand it - and in that case the name is pretty self explanatory

anyway thanks for writing that out!